### PR TITLE
add github action for tests

### LINKED ISSUE #614: Travis CI free usage ends Dec 3; mozilla repos should switch to other CI platforms
We're opening this issue because your project has used Travis CI within the last 6 months. If you have already migrated off it, you can close and ignore this issue.

Travis CI is ending free builds on public repositories. travis-ci.com [stopped providing](https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing)them in early November, and travis-ci.org [will stop](https://mailchi.mp/3d439eeb1098/travis-ciorg-is-moving-to-travis-cicom) after December 31, 2020. To avoid disruptions to your workflows, you must migrate to another CI service.

For production use cases, we recommend switching to [CircleCI](https://circleci.com/). This service is already widely used within Mozilla. There is a guide to migrating from Travis CI to CircleCI available [here](https://circleci.com/docs/2.0/migrating-from-travis/).

For non production use cases, we recommend either CircleCI or [Github Actions](https://github.com/features/actions). There is a guide to migrating from Travis CI to Github Actions available [here](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/migrating-from-travis-ci-to-github-actions). Github Actions usage within Mozilla is new, and you will have to work with our github administrators to enable specific actions following [this process](https://wiki.mozilla.org/GitHub#How_do_I_hook_up_a_new_GitHub_Actions_or_3rd_party_application_to_a_repository_in_the_mozilla_org.3F).

If you have any questions, reach out in `#github-admin:mozilla.org` on [matrix](https://chat.mozilla.org/#/room/#github-admin:mozilla.org).

### PR DESCRIPTION
**Pull request by salilmishra23**
This PR adds github actions for tests, resolves #614.

It currently fails for `tests/remote/test_firefox.py::test_tinderbox_scraper` test as for some arguments the request timesout (Given enough time it returns HTTP 504). 
It is also happening locally. 
Logs - https://github.com/mozilla/mozdownload/actions/runs/3041139681/jobs/4897959420#step:5:366
```
{'branch': 'mozilla-central', 'platform': 'linux64'},
{'branch': 'mozilla-central', 'platform': 'win32'},
{'branch': 'mozilla-central', 'platform': 'win64'},
```

Let me know if you have suggestions.

### CODE REVIEW

**Review by whimboo — CHANGES_REQUESTED** [review]
It's great to see that we can make progress here! Some suggestions regarding to your changes:

1. Please see my inline comment so we can have a broader coverage across differen Python releases.
2. Also remove the configurations for Travis CI and AppVeyour which are not necessary anymore given that GitHub actions now handle all platforms.
3. If we can extend the timeout for the remote tinderbox tests then I'm happy with a change there as well. Otherwise feel free to mark them as skipped for now so that we have a passing test suite. We can check for improvements in a follow-up issue.

Thanks!

**Comment by salilmishra23** [comment]
Thanks for the suggestion. I have added more python version for `ubuntu-latest`, let me know if you want the tests to run on all the versions for the all the OSs.

The tests are failing on Python 3.10 due to `wptserve v3.0` being pinned on `h2 v3.0.1` which is not compatible with Python 3.10. There hasn't been a new release from them(`wptserve`) since 2020. [1]

[1] https://pypi.org/project/wptserve/

**Comment by whimboo** [comment]
That's great! Thanks a lot for the update!

> Thanks for the suggestion. I have added more python version for `ubuntu-latest`, let me know if you want the tests to run on all the versions for the all the OSs.

It would probably be enough for now to only use a single Python version for MacOS and Windows. Maybe we use the last official release for it?

> The tests are failing on Python 3.10 due to `wptserve v3.0` being pinned on `h2 v3.0.1` which is not compatible with Python 3.10. There hasn't been a new release from them(`wptserve`) since 2020. [1]

Would you mind filing an issue for that at https://github.com/web-platform-tests/wpt/issues and referencing here and the yml file? We might have to ignore Python 3.10 for now until it got fixed.

> Other possible improvements:
>     1. You can also add `dependabot` in the repository settings, and we can remove the `.pyup.yaml`.

Lets do that. But as I can read in the [GitHub documentation](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuring-dependabot-version-updates#enabling-github-dependabot-version-updates) it would require a [`dependabot.yml`](https://github.com/mozilla/mozregression/blob/main/.github/dependabot.yml) to be added.

>     2. Getting `coveralls` to work as done by `mozregression`.     
>        * Adding Github token to the github action job and `coveralls --service=github` to the `tox.ini` - [Link](https://github.com/mozilla/mozregression/blob/4f002a750e457fac26fbc0c24adbc85189288acd/.github/workflows/build.yml#L65&L94)
>        * Adding `.coveralls.yml` file as done by them - [Link](https://github.com/mozilla/mozregression/blob/4f002a750e457fac26fbc0c24adbc85189288acd/.coveralls.yml)

Please add whatever you can to the appropriate files. Once the PR is merged I can update the repository settings.

**Comment by salilmishra23** [comment]
Hey @whimboo, the PR is ready for review.

File: requirements/tests.txt [hunk 1]
```diff
@@ -1,5 +1,6 @@
-coveralls==1.8.1
-pytest==4.6.3
-pytest-cov==2.7.1
-pytest-mock==1.10.4
+coveralls==3.3.1
+pytest==7.1.3
+pytest-cov==3.0.0
+pytest-mock==3.8.2
+PyYAML==6.0
```
Comment (whimboo) [inline requirements/tests.txt:5]: Could you please explain why exactly this import is needed? None of our code uses this package and if it's coming from a dependency it should be automatically be installed.

File: tests/remote/test_firefox.py [hunk 2]
```diff
@@ -101,6 +101,7 @@ def test_daily_scraper(tmpdir, args):
     mozdownload.DailyScraper(destination=tmpdir, **args)
 
 
+@pytest.mark.skip(reason="Fails with network timeout for some parameters")
```
Comment (whimboo) [inline tests/remote/test_firefox.py:104]: For mozilla-central tests I do not really expect such a failure. Do you have more details like a formerly failing CI job?

**Review by whimboo — CHANGES_REQUESTED** [review]
This looks great. Just two remaining questions. Once these are solved I would land the code and setup the repository to handle the dependabot and coveralls services.

Please also update the README.md file and exchange the pyup badge with dependabot as well.

File: requirements/tests.txt (hunk 1 above)
Comment (salilmishra23) [reply requirements/tests.txt:5]: you're right! Didn't check the package details then, installing coveralls with extra package should resolve it. Will send a fix soon.
https://github.com/TheKevJames/coveralls-python/blob/master/setup.py#L43

**Comment by salilmishra23** [comment]
Dependabot doesn't seem to have a badge yet dependabot/dependabot-core#1912, so I will remove the pyup badge.

**Review by whimboo — CHANGES_REQUESTED** [review]
Thanks a lot for the last update! There is just one small thing and then we seem to be ready to get this PR merged.

**Comment by salilmishra23** [comment]
Created the issue mozilla/mozdownload#625, once the PR is merged, I will update the description with the `skip` reference.

**Comment by whimboo** [comment]
Thanks for filing the issue. I've updated the skip reason with a link to the newly filed issue. If tests are all passing I'm going ahead and merge this PR.

**Review by whimboo — APPROVED** [review]
Tests are all passing. As such lets get this landed and if something is still missing we can follow-up on. Again, thanks a lot for working on this PR and bringing the project back to life!

**Pull request merged**